src/cross_section/src/cross_section.cpp

```diff
@@ -163,50 +163,46 @@ bool V2Lesser(glm::vec2 a, glm::vec2 b) {
   return a.x < b.x;
 }
 
-double Cross(glm::vec2 a, glm::vec2 b) { return a.x * b.y - a.y * b.x; }
-
-double SquaredDistance(glm::vec2 a, glm::vec2 b) {
-  auto d = a - b;
-  return glm::dot(d, d);
-}
-
-bool IsCw(glm::vec2 a, glm::vec2 b, glm::vec2 c) {
-  double lhs = Cross(a - c, b - c);
-  double rhs = 1e-18 * SquaredDistance(a, c) * SquaredDistance(b, c);
-  return lhs * std::abs(lhs) <= rhs;
-}
-
-void HullBacktrack(const SimplePolygon& pts, const int idx,
-                   std::vector<int>& keep, const int hold) {
-  const int stop = keep.size() - hold;
-  int i = 0;
-  while (i < stop && !IsCw(pts[idx], pts[keep[keep.size() - 1]],
-                           pts[keep[keep.size() - 2]])) {
-    keep.pop_back();
-    i++;
+void HullBacktrack(const glm::vec2& pt, std::vector<glm::vec2>& stack) {
+  auto sz = stack.size();
+  while (sz >= 2 && CCW(stack[sz - 2], stack[sz - 1], pt, 1e-18) <= 0) {
```

Comment:
glm::vec2 is two floats, so they're not usually precise down to 1e-18. Since you aren't making use of the colinear case, I think it's safe to just the tolerance to 0 here.

Reply:
The precision value came from the previous code that was there. I have no idea what it should be.
However, the code IS making use of the colinear case as both colinear and CW points are dropped.

Reply:
@elalish I need a hint here... is it OK to just use the zero, or should I use something else.
The code really is making use of the colinear case.

Reply:
It's really not - at 1e-18, the collinear case is either never being hit, or if it is, it's garbage. Agreed the same error was in the previous code. Changing it to zero shouldn't make any difference, but will make the code less confusing to read.

Reply:
Oh, I see where 1e-18 came from now: he squared both sides, so he was really using a tolerance of 1e-9. I'm still not really sure what that tolerance does, since there's no third case for collinear in the algorithm. You might try a test with a bunch of collinear points to see what difference it makes. I would guess even if you switched from `<= 0` to `< 0` it wouldn't matter much. Probably a *very* slightly concave and convex line of points would be a good test.

Reply:
@elalish OK, after looking at numerous implementations of both Graham and Andrew/Monotone algorithms, I only
found that bothered with it at all. Looking strictly at how the algorithm works, you might actually break things
if you don't use 0. So I used zero and  pushed.